serveronly/watcher.js

```diff
@@ -0,0 +1,318 @@
+const { spawn } = require("child_process");
+const fs = require("fs");
+const path = require("path");
+const net = require("net");
+const http = require("http");
+const Log = require("../js/logger");
+
+const RESTART_DELAY_MS = 500;
+const PORT_CHECK_MAX_ATTEMPTS = 20;
+const PORT_CHECK_INTERVAL_MS = 500;
+
+let child = null;
+let restartTimer = null;
+let isShuttingDown = false;
+let isRestarting = false;
+let serverPort = null;
+const rootDir = path.join(__dirname, "..");
+
+/**
+ * Get the server port from config
+ * @returns {number} The port number
+ */
+function getServerPort () {
+	if (serverPort) return serverPort;
+
+	try {
+		// Try to read the config file to get the port
+		const configPath = path.join(__dirname, "..", "config", "config.js");
```

Comment:
hardcoded `config.js path`? Maybe [this part](https://github.com/MagicMirrorOrg/MagicMirror/blob/develop/js/app.js#L30) of `js/app.js` should be moved to `js/server_functions,js` and used in both places?